src/plugins/intel_gpu/src/kernel_selector/cl_kernels/slice_ref.cl

```diff
@@ -4,33 +4,89 @@
 
 #include "include/batch_headers/fetch_data.cl"
 
-KERNEL(slice_ref)(const __global INPUT0_TYPE* input, __global OUTPUT_TYPE* output)
+#define BRING_INTO_RANGE(VAL, MAX) \
+    clamp((long)VAL < 0l ? (long)VAL + (long)MAX : (long)VAL, 0l, (long)MAX-1l);
+
+#if INPUT0_DIMS < 5
+#define LOAD_BUFFER(in_prefix, out_name)  \
+    long out_name[INPUT0_DIMS];           \
+    out_name[0] = in_prefix##_VAL0;       \
+    out_name[1] = in_prefix##_VAL1;       \
+    out_name[2] = in_prefix##_VAL2;       \
+    out_name[3] = in_prefix##_VAL3;
+#else
+#define LOAD_BUFFER(in_prefix, out_name)  \
+    long out_name[INPUT0_DIMS];           \
+    out_name[0] = in_prefix##_VAL0;       \
+    out_name[1] = in_prefix##_VAL1;       \
+    out_name[2] = in_prefix##_VAL2;       \
+    out_name[3] = in_prefix##_VAL3;       \
+    out_name[4] = in_prefix##_VAL4;
+#endif
+
+KERNEL(slice_ref)(OPTIONAL_SHAPE_INFO_ARG 
+                  const __global INPUT0_TYPE* restrict input,
+                  START_BUFFER
+                  STEP_BUFFER
+                  AXES_BUFFER
+                  __global OUTPUT_TYPE* restrict output)
 {
-    const uint batch = get_global_id(0);
-    const uint feature = get_global_id(1);
+    LOAD_BUFFER(START, start_buff);
+    LOAD_BUFFER(STEP, step_buff);
+    LOAD_BUFFER(AXES, axes_buff);
+
+    long slice_step[INPUT0_DIMS];
+    long slice_start[INPUT0_DIMS];
+    #pragma unroll
+    for(int i = 0; i < INPUT0_DIMS; ++i) {
+        slice_step[i] = 1;
+        slice_start[i] = 0;
+    }
+
+    #pragma unroll
+    for(int i = 0; i < AXES_BUFFER_SIZE; ++i) {
+        const long axis = axes_buff[i];
+        slice_step[axis] = step_buff[i];
+        slice_start[axis] = start_buff[i];
+    }
+
+    const long output_dim0 = get_global_id(0);
+    const long output_dim1 = get_global_id(1);
```

Comment:
It's better to use uint for `get_global_id()` and indexes calculations, since `long` calculation may be quite time consuming, but uint data range is enough for most of the cases

Reply:
I thought about that. However, it is problematic IMO. First of all, the slice op support int64_t values(I also agree that it is pointless, but that's the definition of that op). So in general it might happen that somebody will one day use values that will fit only into int64. Since this is a reference implementation, it is suppose to work with all possible parameters, I have decided to keep everything in long, since it will always fit. Previously, slice op was silently down casting in64_t to in32, which was bad IMO. If we don't want to support int64_t, then it should be an assert, that  "you have int64_t input parameters, but gpu supports only int32_t". But that is the question of what is that reference implementation. 

About performance, I agree, it may happen that performance might be less - however it all depends on the hardware. In this particular case the degradation shouldn't be much(however, I haven't check it and my claims are based mostly on NVIDIA architectures), since the kernel is memory_bound, so in general case, the bottleneck of this kernel will be memory access pattern , not the offset calculation itself(assuming that you have enough blocks/threads to switch to - it is called high enough occupancy in NVIDIA terms - it should be true since I am talking about general case here, not some edge cases where data is small etc).

So to sum up, taking all the tradeoffs, I have assumed that it is more important for this kernel to be conformant with this slice op def, than being 'optimized', since this kernel is not optimized anyway:) The goal was to add support for dynamic shape, not to optimize it, so I tried to do it in way, that the performance in cases that were already supported(namely, static input) will not be seriously degraded.

Reply:
@pkowalc1 yeah, absolutely agree with your point
Just some additional thoughts:
If there will be the real case where uint32_t data range wouldn't be enough, most likely the plugin will fail much earlier, because of the same limitations in the other kernels :)
As this slice op is going to replace strided_slice (and it is supposed to have at least on par or even better performance), some additional perf can be helpful
But personally I'm okay with long data type